core/src/com/unciv/logic/city/CityResources.kt

```diff
@@ -8,20 +8,43 @@ import com.unciv.models.ruleset.unique.UniqueType
 
 object CityResources {
 
-    fun getCityResources(city: City): ResourceSupplyList {
-        val cityResources = ResourceSupplyList()
+    /** Returns ALL resources, city-wide and civ-wide */
+    fun getResourcesGeneratedByCity(city: City): ResourceSupplyList {
+        val resourceModifers = HashMap<String, Float>()
+        for (resource in city.civ.gameInfo.ruleset.tileResources.values)
+            resourceModifers[resource.name] = city.civ.getResourceModifier(resource)
+
+        val cityResources = getResourcesGeneratedByCityNotIncludingBuildings(city, resourceModifers)
+        addCityResourcesFromUniqueBuildings(city, cityResources, resourceModifers)
+
+        return cityResources
+    }
 
-        val resourceModifer = HashMap<String, Float>()
+
+    /** Only for *city-wide* resources - civ-wide resources should use civ-level resources */
+    fun getCityResourcesAvailableToCity(city: City): ResourceSupplyList {
+        val resourceModifers = HashMap<String, Float>()
         for (resource in city.civ.gameInfo.ruleset.tileResources.values)
-            resourceModifer[resource.name] = city.civ.getResourceModifier(resource)
+            resourceModifers[resource.name] = city.civ.getResourceModifier(resource)
+
+        val cityResources = getResourcesGeneratedByCityNotIncludingBuildings(city, resourceModifers)
+        // We can't use getResourcesGeneratedByCity directly, because that would include the resources generated by buildings -
+        //   which are part of the civ-wide uniques, so we'd be getting them twice!
+        // This way we get them once, but it is ugly, I welcome other ideas :/
+        getCityResourcesFromCiv(city, cityResources, resourceModifers)
+
+        return cityResources
+    }
 
-        getResourcesFromTiles(city, resourceModifer, cityResources)
 
-        getResourceFromUniqueImprovedTiles(city, cityResources, resourceModifer)
+    private fun getResourcesGeneratedByCityNotIncludingBuildings(city: City, resourceModifers: HashMap<String, Float>): ResourceSupplyList {
```

Comment:
Why not including buildings? You can check buildings of just this city with `city.getMatchingLocalOnlyUniques`

Side note, probably accident of mine, but `city.getMatchingLocalOnlyUniques` and `getLocalMatchingUniques` reads as identical. One needs removing

Reply:
Ah, nevermind. I see what's going on. Still, using local only uniques here is ideal. The later calls can easily check the uniques for local effects to avoid doubling